pipeline.cpp

```diff
@@ -153,6 +153,10 @@ void DrawTriangle(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes*
                                                           attrs[1].attr[0],attrs[2].attr[0]);
                 interpolatedAttribs.attr[1] = z * interoplate(y, x, triangle, attrs[0].attr[1],
                                                           attrs[1].attr[1],attrs[2].attr[1]);
+                interpolatedAttribs.attr[2] = z * interoplate(y, x, triangle, attrs[0].attr[2],
```

Comment:
You should just keep track of how many attributes you have so you can do this in a loop depending on your number of attributes